pkg/frontend/v2/frontend.go

```diff
@@ -192,6 +194,8 @@ func (f *Frontend) RoundTripGRPC(ctx context.Context, req *httpgrpc.HTTPRequest)
 		// even if this goroutine goes away due to client context cancellation.
 		enqueue:  make(chan enqueueResult, 1),
 		response: make(chan *frontendv2pb.QueryResultRequest, 1),
+
+		retryOnTooManyOutstandingRequests: f.schedulerWorkers.getWorkersCount() > 0,
```

Comment:
If multiple query schedulers have queues full, then retry may make queueing even worse? A request will retry forever until it is not discarded by the queue.

Reply:
I think the retry can still be useful. Let's add a feature flag?

Reply:
`retries := f.cfg.WorkerConcurrency + 1` limits the maximum number of retries.

A feature flag sounds good.